pyresample/geometry.py

```diff
@@ -1175,60 +1178,16 @@ def get_val(val, sub_val, max):
         if dtype is None:
             dtype = self.dtype
 
-        # create coordinates of local area as ndarrays
+        target_x = np.arange(self.x_size, dtype=dtype) * self.pixel_size_x + self.pixel_upper_left[0]
+        target_y = np.arange(self.y_size, dtype=dtype) * -self.pixel_size_y + self.pixel_upper_left[1]
         if data_slice is None or data_slice == slice(None):
-            # Full slice
-            rows = self.y_size
-            cols = self.x_size
-            row_start = 0
-            col_start = 0
+            pass
+        elif isinstance(data_slice, slice):
+            target_y = target_y[data_slice]
         else:
-            if isinstance(data_slice, slice):
-                # Row slice
-                row_start = get_val(data_slice.start, 0, self.y_size)
-                col_start = 0
-                rows = get_val(
-                    data_slice.stop, self.y_size, self.y_size) - row_start
-                cols = self.x_size
-            elif isinstance(data_slice[0], slice) and isinstance(data_slice[1], slice):
-                # Block slice
-                row_start = get_val(data_slice[0].start, 0, self.y_size)
-                col_start = get_val(data_slice[1].start, 0, self.x_size)
-                rows = get_val(
-                    data_slice[0].stop, self.y_size, self.y_size) - row_start
-                cols = get_val(
-                    data_slice[1].stop, self.x_size, self.x_size) - col_start
-            elif isinstance(data_slice[0], slice):
-                # Select from col
-                is_1d_select = True
-                row_start = get_val(data_slice[0].start, 0, self.y_size)
-                col_start = get_val(data_slice[1], 0, self.x_size)
-                rows = get_val(
-                    data_slice[0].stop, self.y_size, self.y_size) - row_start
-                cols = 1
-            elif isinstance(data_slice[1], slice):
-                # Select from row
-                is_1d_select = True
-                row_start = get_val(data_slice[0], 0, self.y_size)
-                col_start = get_val(data_slice[1].start, 0, self.x_size)
-                rows = 1
-                cols = get_val(
-                    data_slice[1].stop, self.x_size, self.x_size) - col_start
-            else:
-                # Single element select
-                is_single_value = True
-
-                row_start = get_val(data_slice[0], 0, self.y_size)
-                col_start = get_val(data_slice[1], 0, self.x_size)
-
-                rows = 1
-                cols = 1
+            target_y = target_y[data_slice[0]]
+            target_x = target_x[data_slice[1]]
```

Comment:
Wow this whole block of code got a lot smaller.

Reply:
Yes, but the downside is that the arange is called before the slicing is performed

Reply:
Couldn't this be `da.arange`?

Reply:
Then we would need to put dask as a requirement.

Reply:
Ah sorry, github's diff tool was making it look like this was the `_dask` method.